src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops.cpp

```diff
@@ -81,6 +91,12 @@ bool PerformanceImplicitGemmForwardV4R4Xdlops::SetNextValue()
     {
         // list performance parameters in reverse order, in order for tuning to iterate over the
         // range in normal order
+        if(miopen::IsEnabled(
+               MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_FWD_V4R4_XDLOPS_ADD_VECTOR_LOAD_GEMMN_TUNE_PARAM{}))
+        {
+            if(!NextTwoPower<1, 8>(GemmBThreadDataPerRead_GemmN))
+                break;
+        }
         if(!NextFlag<false, true>(GemmBThreadCopyMoreGemmKPack))
             break;
         if(!NextFlag<false, false>(GemmAThreadCopyMoreGemmK))
```

Comment:
Hi artem, 
what do you mean by 'semantics'? do you mean the usage scenario? if yes, pls refer to the comment of this pr.
However, I do see I should also deal with different kind of data type, for fp32, it's <1,4>, for fp16, it's <1,8> here. I'll fix it.

Reply:
NextFlag<A, B> is same as NextTwoPower, it will loop over A -> B - > A...

Reply:
This construction is ambiguous. `false` is the minimum and `false` is the maximum. How will the `NextFlag<min,max>()` function accept such a value? If the function would think that `false` on input is maximum, then it will increment it, and return an overflow indicator (`true`). If the function would think that `false` is minimum, it will increment it to the maximum, and does not report overflow to the caller (return `false`). To understand how all this works, one needs need to go into the implementation of `NextFlag()`.

However, this code is no different from a regular assignment. Moreover, there is no reason for doing this in in `SetNextValue()`. I would say this should be done exactly once, during initialization.

I already see that these constructs have proliferated in our code like rabbits. I think thast all of them must be eliminated.

Reply:
Hi atamazov,
Intentionally, NextFlag<> is similar as 'NextTwoPower', which will make a value to be set 'false' and 'true' while tunning. if it's NextFlag<false, false>, it will make a param untunable, but only set to 'false'. it's functionally same as just set its to 'false'.
However, I guess, original author, here, would try to make it tunable, and this it may be changed to tunable param someday how. 
@asroy , what do you think? leave it here or simply remove it from setNextValue?

Reply:
> if it's NextFlag<false, false>, it will make a param untunable, but only set to 'false'. 

Of course I know.

> it's functionally same as just set its to 'false'.

No, it is not the same because is also returns a wraparound indicator. The reader must read the definition of `NextFlag()` to find out which value the function would return in the `<false,false>` case.

> However, I guess, original author, here, would try to make it tunable, and this it may be changed to tunable param someday how.

This is over-design. 

There are an infinite number of "what if ..." cases in the potential future. The code that supports such cases has no immediate functional purpose. That confuses the reader who must guess why this was done. The more such code we have, the more difficult is to read and maintain the library. That as why we have to avoid this, or, in _exceptional_ cases, some explanatory comment must be added.

WRT this specific case, i.e. "idle" PerfConfig data members. Basically we should completely remove those. However, we can keep them, --- ___if it is highly likely___ that these will be needed in the future. This _exception_ is the simplest solution to make today's perf-db compatible with a future version of the code.

How this should be done?
- Put an explanatory comment near the definition of an "idle" member of PerformanceConfig.
- Initialize it but do not use in `Next...()`

Reply:
@jane-zxy You can follow @atamazov 's suggestion 1) remove this line from SetNextValue(), 2) comment in ```PerformanceImplicitGemmForwardV4R4Xdlops``` that this tuning parameter is a fixed value currently.